Sources/CNAMEPublishPlugin/CNAMEPublishPlugin.swift

```diff
@@ -13,11 +13,11 @@ public extension Plugin {
     /// output directory.
     ///
     /// - Parameter domainNames: A list of domain names to use for the website.
-    static func generateCNAME(with domainNames: String...) -> Self {
+    static func generateCNAME(with domainName: String, _ otherDomainNames: String...) -> Self {
```

Comment:
@a2 

Thank you for your contribution and for using this plugin. Correct me if I'm wrong, I have a couple of concerns about this change:

1. From the title of the pull request, the goal is to prevent an empty list for generating a file. But currently, we already do that with this block of code that was removed:
    ```swift
    guard !domainNames.isEmpty else { return }
    ```
    We can add a unit test to the existing code base that asserts when no domain names are provided, that the file isn't generated.
2. For the method signature change, are there benefits of using two parameters vs. using one variadic parameter? Using the variadic parameter allows us to use the method in a couple of ways:
    ```swift
    try TestWebsite().publish(at: Self.testDirPath, using: [
        .installPlugin(.generateCNAME(with: "test.io", "www.test.io")),
        .installPlugin(.generateCNAME(with: "Test.io"))
     ])
    ```
    I can see a use case where the API user would want to pass in an array of domain names which currently doesn't work with the current implementation. If we want to support that, I would suggest creating a new method that takes in an array of domain names and then have the existing method call the new method. Here is an example:
    ```swift
    static func generateCNAME(with domainNames: String...) -> Self {
        generateCNAME(with: domainNames)
    }

    static func generateCNAME(with domainNames: [String]) -> Self {
        Plugin(name: "Generate CNAME for custom domain names") { context in
            guard !domainNames.isEmpty else { return }
            
            let fileContent = domainNames.joined(separator: "\n")
            
            try context.createOutputFile(at: "CNAME").write(fileContent)
        }
    }
    ```
What are your thoughts on this?

Reply:
A `String...` argument allows the user to provide an empty list, if I remember correctly. That's why I wanted to add an additional `String` parameter before it. With this, the compiler can check that there is always at least one argument. Otherwise, as you mentioned the `guard !domainNames.isEmpty else { return }` check, I would prefer not to use that because it's a runtime check for something that the compiler is also able to check.  

I also like the idea of a `[String]` version as well, in case the list of CNAME domains is programmatically generated. I believe the length of a `String...` is not changeable at runtime.

Reply:
Another thing to consider(something that I didn't think of initially) is if the provided first or any of the other domain names are empty strings. For that, we can throw an error from the plugin indicating to the user that a domain name is required. Based on our discussion, would you like to make the proposed changes(If you have some free time)? Or would you like me to make the changes in a new pull request?

Reply:
Feel free to make changes to branch; I enabled "Allow edits from maintainers". I won't have the bandwidth to get to this very soon.